### PR TITLE
Set and clear XML_CATALOG_FILES on activation and deactivation, respectively (take 3)

### PR DESCRIPTION
**Pull request by hattne**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

From https://github.com/conda-forge/libxml2-feedstock/pull/87: avoid non-POSIX shell constructs, `{,de}activivate.sh` pass `shellcheck` under `/bin/sh`.

### CODE REVIEW

**Comment by gillins** [comment]
@hattne this is still a draft - is it ready for review?

**Comment by hattne** [comment]
It's been running without any issues here for a couple of days (but that was true for the previous PRs as well).

Ready for review.

**Comment by gillins** [comment]
@ocefpaf do you want to check this one?

**Comment by ocefpaf** [comment]
Let's try it out! Thanks @hattne and @gillins!

**Comment by jaimergp** [comment]
[Seeing unbound variables errors again](https://github.com/conda/conda-libmamba-solver/actions/runs/4433619352/jobs/7778828446#step:6:2847) :(

```
/Users/runner/miniconda3/envs/conda-test-env/etc/conda/deactivate.d/libxml2_deactivate.sh: line 3: xml_catalog_files_libxml2: unbound variable
```

File: recipe/deactivate.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#! /bin/sh
 
 if test -n "${xml_catalog_files_libxml2}"; then
```
Comment (jaimergp) [inline recipe/deactivate.sh:3]: Maybe this just needs `"${xml_catalog_files_libxml2:-}"?

**Comment by jaimergp** [comment]
See https://github.com/conda-forge/admin-requests/pull/694 @ocefpaf

**Pull request merged**